### ISSUE TITLE
Exception for missing CUDA dll

### ISSUE THREAD

**Issue by OsirisTerje** [mathnet/mathnet-numerics-native#18]
I am using Mathnet.Numerics in a library, and the app that uses this package, also includes the mathnet.numerics.  When I try to debug the app, I get a 

Exception thrown: 'System.DllNotFoundException' in MathNet.Numerics.dll

Additional information: Unable to load DLL 'MathNet.Numerics.CUDA.dll': The specified module could not be found. 

There is no CUDA dll in the package, and I assume this is a native provider, but I have not enabled anything.  So, why is this coming up?  How do I disable this?

**Comment by cdrnet** [comment]
To my understanding this should no longer happen since v3.11. Which version are you using?

You can safely continue/skip the exception, it is caught internally (your debugger is likely configured to always pause when this particular exception is thrown, even if it is caught).

**Comment by OsirisTerje** [comment]
Version 3.11.1
And after this one I get a 
Exception thrown: 'System.NotSupportedException' in MathNet.Numerics.dll

Additional information: Cuda Native Provider not found.

And then followed by similar pair ones for MKL
Exception thrown: 'System.DllNotFoundException' in MathNet.Numerics.dll

Additional information: Unable to load DLL 'MathNet.Numerics.MKL.dll': 
Exception thrown: 'System.NotSupportedException' in MathNet.Numerics.dll

Additional information: MKL Native Provider not found.

And then

Exception thrown: 'System.DllNotFoundException' in MathNet.Numerics.dll

Additional information: Unable to load DLL 'MathNET.Numerics.OpenBLAS.dll':

with
Exception thrown: 'System.NotSupportedException' in MathNet.Numerics.dll

Additional information: OpenBLAS Native Provider not found.

**Comment by ebyrne** [comment]
We're on 3.12, just upgraded, and we're seeing the same thing.  (3 separate sets of first chance exceptions - both DllNotFound and NotSupported for each.)  Is that the expected behavior currently?  Would a stack help?

If I might humbly suggest, for something like this a custom exception rather than something as common as NotSupportedException would go a long way towards improving the debugging experience since folks would often be hesitant to disable all NotSupportedException's.

**Comment by mguntli** [comment]
Version 3.12.0.0 with the same behaviour

**Comment by darrenreynolds** [comment]
v3.13.1. I have exactly the same sequence of errors. Is there something I can do to resolve them?

**{"Unable to load DLL 'MathNet.Numerics.CUDA.dll': The specified module could not be found...}
Cuda Native Provider not found.

{"Unable to load DLL 'MathNet.Numerics.MKL.dll': The specified module could not be found...}
MKL Native Provider not found.

{"Unable to load DLL 'MathNET.Numerics.OpenBLAS.dll': The specified module could not be found...}
OpenBLAS Native Provider not found**

**Comment by cdrnet** [comment]
Just to clarify: none of these are exceptions that ever reach your code, they only show up on debugging when the debugger is configured to break on all exceptions (which is inconvenient), even if caught. Is this correct?

**Comment by cdrnet** [comment]
PS: Native providers are no longer developed in this repo, but instead in the [main repository](https://github.com/mathnet/mathnet-numerics/) directly.

**Comment by darrenreynolds** [comment]
Yes, that is correct. It is a matter of convenience. You'll appreciate why
some developers have 'break on all errors' enabled.

**Comment by cdrnet** [comment]
I do appreciate it. I'll see whether we can improve this further.
